core/DataAccess/Model.php

```diff
@@ -238,13 +246,36 @@ public function updateArchiveAsInvalidated($archiveTable, $idSites, $allPeriodsT
             }
         }
 
-        $fields = ['idarchive', 'name', 'report', 'idsite', 'date1', 'date2', 'period', 'ts_invalidated'];
-
-        Db\BatchInsert::tableInsertBatch(Common::prefixTable('archive_invalidations'), $fields, $dummyArchives);
+        if (!empty($dummyArchives)) {
+            $fields = ['idarchive', 'name', 'report', 'idsite', 'date1', 'date2', 'period', 'ts_invalidated'];
+            Db\BatchInsert::tableInsertBatch(Common::prefixTable('archive_invalidations'), $fields, $dummyArchives);
+        }
 
         return count($idArchives);
     }
 
+    private function getExistingInvalidations($idSites)
+    {
+        $table = Common::prefixTable('archive_invalidations');
+
+        $sql = "SELECT idsite, date1, date2, period, name, COUNT(*) as count FROM `$table`
+                 WHERE idsite IN (" . implode(',', $idSites) . ")
```

Comment:
btw we would need to ignore any archive currently being in progress? `status=1`?

Reply:
Yes, thought of that too 👍

Reply:
@diosmosis be great to add the status check. I guess it should be `status=0` or `status!=1`